crates/walrus-service/src/node/events/event_processor.rs

```diff
@@ -652,18 +652,36 @@ impl EventProcessor {
 
         event_processor.update_checkpoint_cache(current_checkpoint);
 
-        let latest_checkpoint = retry_client.get_latest_checkpoint_summary().await?;
-        if current_checkpoint > latest_checkpoint.sequence_number {
-            tracing::error!(
-                current_checkpoint,
-                ?latest_checkpoint,
-                "Current store has a checkpoint that is greater than latest network checkpoint! \
-                    This is especially likely when a node is restarted running against a newer \
-                    localnet, testnet or devnet network."
+        let latest_checkpoint = match retry_client.get_latest_checkpoint_summary().await {
```

Comment:
Nit: this function is getting very long. Can you put `fetching lag + catch up` into a separate function?

Reply:
done